dotCMS/src/integration-test/java/com/dotcms/contenttype/business/FieldAPITest.java

```diff
@@ -1035,13 +1036,64 @@ public void testSaveFieldVariable_KeyWithSpecialChars_ShouldSucceed(final String
             fieldAPI.save(variable, user);
 
             boolean anyMatch = field.fieldVariables().stream()
-                  .anyMatch((var)->var.key().equals(fieldVarKey));
+                    .anyMatch((var)->var.key().equals(fieldVarKey));
 
             Assert.assertTrue("Incorrect var key", anyMatch);
         } finally {
             contentTypeAPI.delete(type);
         }
     }
 
+    @DataProvider
+    public static Object[] dataProviderGraphQLReservedNames() {
+        return InterfaceType.RESERVED_GRAPHQL_FIELD_NAMES.toArray();
+    }
+
+    @Test
+    @UseDataProvider("dataProviderGraphQLReservedNames")
+    public void test_SaveFieldWithReservedGraphqlName_ShouldSuffixConsecutiveToVariable(
+            final String fieldName)
+            throws DotSecurityException, DotDataException {
+
+        final ContentType type = new ContentTypeDataGen().nextPersisted();
+        try {
+            Field field1 = FieldBuilder.builder(TextField.class)
+                    .name(fieldName)
+                    .contentTypeId(type.id())
+                    .indexed(false)
+                    .listed(false)
+                    .fixed(true)
+                    .build();
+            field1 = fieldAPI.save(field1, user);
+
+            Assert.assertNotNull(field1);
+            Assert.assertTrue(UtilMethods.isSet(field1.variable()));
+            Assert.assertNotEquals(fieldName, field1.variable());
+
+            // let's create a new field to make sure it's getting a new variable
+
+            Field field2 = FieldBuilder.builder(TextField.class)
+                    .name(fieldName)
+                    .contentTypeId(type.id())
+                    .indexed(false)
+                    .listed(false)
+                    .fixed(true)
+                    .build();
+            field2 = fieldAPI.save(field2, user);
+
+            Assert.assertNotNull(field2);
+            Assert.assertTrue(UtilMethods.isSet(field2.variable()));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertTrue' due to existing static import 'org.junit.Assert.assertTrue'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4926848)